src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseExportImportResource.java

```diff
@@ -267,7 +268,7 @@ public ResponseEntity<Resource> exportInstructorExercise(@PathVariable long exer
         long start = System.nanoTime();
         Path path;
         try {
-            path = programmingExerciseExportService.exportProgrammingExerciseInstructorMaterial(programmingExercise, Collections.synchronizedList(new ArrayList<>()));
+            path = programmingExerciseExportService.exportProgrammingExerciseForDownload(programmingExercise, Collections.synchronizedList(new ArrayList<>()));
```

Comment:
Why can't this be a `concurrentList` instead of  a `synchronizedList`?

Reply:
tbh, I didn't introduce this but @krusche did in #7165.

Reply:
I found this online to be the best solution for concurrent threads. Tbh, I do not really know the differences between synchronizedList and synchronizedList

Reply:
as long as it works, and there are no big disadvantages for synchronizedList, I would leave it

Reply:
concurrentList allows multiple threads to perform R/W-OPs concurrently. For synchronizedList this is limited to only one thread at a time.
Leaving it like this is fine with me 👍